juju/juju-qa-jenkins
# [JUJU-3488] Add option to generate suites with minimal version condition

Sometimes our integration tests test a recently added feature, meaning
we only wish to run the test for versions beyond some specific minor
release

Also extract wait-for-cloud-init and -prepare-integration-test out of
run-integration-test since the envvars collected by
prepare-integration-test are required for this conditional step

Add an option to juju.config which, if set, wraps our integration test
builder in a conditional clause

Implement this for the new ssh allow firewall test (test-firewall)

Also, it seems someone added some tests to the controller suite but
forgot to add them to Jenkins. Add them here as a flyby

## QA Steps

https://jenkins.juju.canonical.com/job/test-firewall-test-firewall-ssh-aws/

Verify this pipeline succeed for Juju versions on both sides of 3.2

- For 3.2 or beyond, the integration test should run
- For versions prior to 3.2, the pipeline should succeed but not run the integration test
